src/tiered/tiered_handle.c

```diff
@@ -65,8 +69,16 @@ __tiered_init_tiers(WT_SESSION_IMPL *session, WT_TIERED *tiered, WT_CONFIG_ITEM
         WT_ERR(__wt_buf_fmt(session, tmp, "%.*s", (int)ckey.len, ckey.str));
         __wt_verbose(
           session, WT_VERB_TIERED, "INIT_TIERS: tiered URI dhandle %s", (char *)tmp->data);
-        WT_ERR(__tiered_dhandle_setup(
-          session, tiered, WT_TIERED_INDEX_INVALID, (const char *)tmp->data));
+        WT_SAVE_DHANDLE(session,
+          ret = __tiered_dhandle_setup(
+            session, tiered, WT_TIERED_INDEX_INVALID, (const char *)tmp->data));
+        WT_ERR(ret);
+    }
+    local_tier = &tiered->tiers[WT_TIERED_INDEX_LOCAL];
+    if (local_tier->name == NULL) {
+        WT_ERR(__wt_tiered_name(
+          session, &tiered->iface, tiered->current_id, WT_TIERED_NAME_LOCAL, &local_tier->name));
+        F_SET(local_tier, WT_TIERS_OP_READ | WT_TIERS_OP_WRITE);
```

Comment:
Not sure if it is worth it, but `tiered_create_local` has very similar 5 lines setting those flags and getting the name, etc. I'm not sure if it is worth trying to refactor both. `tiered_create_local` does a bunch of other stuff in between creating the name and setting the flags.